src/bika/lims/validators.py

```diff
@@ -1550,6 +1551,55 @@ def __call__(self, value, **kwargs):
             ))
 
 
+class UniqueReferenceSampleIDValidator(object):
+    """
+    Ensures that no existing object in the system has an ID matching the value
+    of the field to which this validator is applied.
+    """
+    implements(IValidator)
+    name = "unique_referencesample_id_validator"
+
+    def __call__(self, value, **kwargs):
+        instance = kwargs["instance"]
+
+        # skip if no value provided
+        if not value:
+            return
+
+        # skip if the value matches with object's current id
+        if instance.getId() == value:
+            return
+
+        # check if the id is valid
+        parent = api.get_parent(instance)
+        if not api.is_valid_id(value, container=parent):
+            return _t(_(
+                u"validator_referencesample_id_invalid",
+                default=u"The Reference Sample ID is invalid. Ensure it "
+                        u"contains only letters, numbers, underscores ('_'), "
+                        u"or hyphens ('-'), and verify that no other "
+                        u"Reference Sample with the same ID already exists.",
+            ))
+
+        # check if a reference sample with this id exists already
+        uid = api.get_uid(instance)
+        cat = api.get_tool(SENAITE_CATALOG)
+        brains = cat(portal_type="ReferenceSample")
```

Comment:
Why not simply search for this ID directly using the `getId` index?

Reply:
The validator is named `UniqueReferenceSampleIDValidator`, so it makes sense to limit its scope to `ReferenceSample` objects. If I were to check all existing IDs across the system, the validator's name would need to be more generic, such as `UniqueIDValidator`. However, this approach would require searching through all catalogs, not just the `SENAITE_CATALOG`. I considered the potential overhead this broader search would introduce and decided to restrict the validation to `ReferenceSample` objects to maintain both consistency and efficiency.

Reply:
I actually meant to use the `getId` additionally, e.g.:
```python
brains = cat(portal_type="ReferenceSample", getId=value)
```

Reply:
Ouch! :sweat_smile: d8100d2c8ff38ef7dff978221b33e40598cc24f6